silhouette/app/com/mohiva/play/silhouette/api/services/AuthenticatorService.scala

```diff
@@ -20,9 +20,52 @@
 package com.mohiva.play.silhouette.api.services
 
 import com.mohiva.play.silhouette.api.{ Authenticator, LoginInfo }
+import play.api.libs.concurrent.Execution.Implicits._
 import play.api.mvc.{ Headers, RequestHeader, Result }
 
 import scala.concurrent.Future
+import scala.language.implicitConversions
+import scala.util.Try
+
+/**
+ * A marker result which indicates that an operation on an authenticator was processed and
+ * therefore it shouldn't updated automatically.
+ *
+ * Due the fact that the update method gets called on every subsequent request to update the
+ * authenticator related data in the backing store and in the result, it isn't possible to
+ * discard or renew the authenticator simultaneously. This is because the "update" method would
+ * override the result created by the "renew" or "discard" method, because it will be executed
+ * as last in the chain.
+ *
+ * As example:
+ * If we discard the session in a Silhouette action then it will be removed from session. But
+ * at the end the update method will embed the session again, because it gets called with the
+ * result of the action.
+ *
+ * @param result The result to wrap.
+ */
+class AuthenticatorResult(result: Result) extends Result(result.header, result.body, result.connection)
+
+/**
+ * The companion object.
+ */
+object AuthenticatorResult {
+
+  /**
+   * Converts a [[Result]] into [[AuthenticatorResult]]
+   */
+  implicit def toAuthenticatorResult(result: Result): AuthenticatorResult = apply(result)
+  implicit def toAuthenticatorResultF(result: Future[Result]): Future[AuthenticatorResult] = result.map(apply)
+  implicit def toAuthenticatorResultT(result: Try[Result]): Try[AuthenticatorResult] = result.map(apply)
+
```

Comment:
>  we would not return a real Result and the user must unwrap the result before he sends it to the client.

Why you need to return a Result? I think the result should get unwrapped by the ActionBuilder after it notices the wrapper & skips the auto renewal process.  In the normal flow, client code returns a Result whether or it uses the Silhouette ActionBuilders.  In cases where client code needs to modify the Authenticator it should be wrapped in Silhouette ActionBuilders and return the marker wrapper class to signal that it wants to opt out of the normal renewal process.

> loses the marker by doing an authenticatorResult.copy

By sublassing Result you make it very easy for the developer to call withSession or .withHeader on the marker wrapper, which will strip the marker wrapper off which in almost all cases will not be the intended result.